share/wake/lib/system/remote_cache_runner.wake

```diff
@@ -0,0 +1,210 @@
+
+                def (
+                    CacheSearchResponseMatch
+                    outputSymlinks
+                    outputDirs
+                    outputFiles
+                    stdoutBlob
+                    stderrBlob
+                    status
+                    runtime
+                    cputime
+                    mem
+                    ibytes
+                    obytes
+                ) = details
+
+                def doDownload ((CacheSearchOutputFile path mode blob): CacheSearchOutputFile) =
+                    rscApiGetFileBlob blob "{input.getRunnerInputDirectory}/{path}" mode
+
+                def blobDownloads =
+                    outputFiles
+                    | map doDownload
+
+                def stdoutDownload = rscApiGetStringBlob stdoutBlob
+                def stderrDownload = rscApiGetStringBlob stderrBlob
+
+                require Pass stdout = stdoutDownload
+                else failWithError "rsc: Failed to download stdout"
+
+                require Pass stderr = stderrDownload
+                else failWithError "rsc: Failed to download stderr"
+
+                # We don't actually care about the result here but we do need to ensure that all
+                # downloads have completed and succeeded before we continue on.
+                require Pass _ =
+                    blobDownloads
+                    | findFail
+                else failWithError "rsc: Failed to download a blob"
+
+                def resolvedOutputs =
+                    outputFiles
+                    | map (_.getCacheSearchOutputFilePath)
+
+                def outputs = resolvedOutputs ++ outputDirs ++ outputSymlinks
+                def predict = Usage status runtime cputime mem ibytes obytes
+                def _ = virtual job stdout stderr status runtime cputime mem ibytes obytes
+                def inputs = map getPathName (input.getRunnerInputVisible)
+
+                Pass (RunnerOutput inputs outputs predict)
+
+            def _ = breadcrumb "Did not find a match"
+
+            # -------------------------------------
+            # --- Insert the job into the cache ---
+            # -------------------------------------
+
+            # Run the job to get the results
+            require Pass output = baseDoIt job (Pass input)
+
+            # Post the job to the server. This is left as 'def _' so that wake won't block progess
+            # on it but it will stil be joined on before wake exits. Regardless of the result this
+            # call should return the passing output to allow the build to continue.
+            def _ = postJob rscApi job wakeroot hashKey input output
+
+            Pass output
+
+    Runner "remote-cache: {name}" score doit
+
+## --- Helper functions ---
+
+# Creates a CacheSearchRequest from the various inputs to a runner
+def mkSearchRequest ((RunnerInput _label cmd vis env dir stdin _res _prefix _ isAtty): RunnerInput) (hidden: String) =
```

Comment:
Can we use `_usage` for the second to last element just to make sure we can easily see which inputs are not being used in the rsc request?